google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerCloudMonitoringExporter.java

```diff
@@ -114,27 +112,19 @@ public CompletableResultCode export(@Nonnull Collection<MetricData> collection)
 
   /** Export client built in metrics */
   private CompletableResultCode exportSpannerClientMetrics(Collection<MetricData> collection) {
-    // Filter spanner metrics. Only include metrics that contain a project and instance ID.
-    List<MetricData> spannerMetricData =
-        collection.stream()
-            .filter(md -> SPANNER_METRICS.contains(md.getName()))
-            .collect(Collectors.toList());
+    // Filter spanner metrics. Only include metrics that contain a valid project.
+    List<MetricData> spannerMetricData = collection.stream().collect(Collectors.toList());
 
     // Log warnings for metrics that will be skipped.
     boolean mustFilter = false;
     if (spannerMetricData.stream()
-        .flatMap(metricData -> metricData.getData().getPoints().stream())
+        .map(metricData -> metricData.getResource())
         .anyMatch(this::shouldSkipPointDataDueToProjectId)) {
       logger.log(
           Level.WARNING, "Some metric data contain a different projectId. These will be skipped.");
       mustFilter = true;
     }
-    if (spannerMetricData.stream()
-        .flatMap(metricData -> metricData.getData().getPoints().stream())
-        .anyMatch(this::shouldSkipPointDataDueToMissingInstanceId)) {
```

Comment:
Curious - why are we skipping this. Did we fix the issue that a instance id could be missing (because we will set "unknown" by default?).

My concern is that if there is a bug that causes the instance id not to be correctly populated, we won't be able to catch that regression.